# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1441232299 14400
#      Wed Sep 02 18:18:19 2015 -0400
# Node ID c36f06ef75392a8ea1f0e96099ac34637aaf6012
# Parent  7fa1970a655de29862d28b361545291b2238c0d6
also clear autoloaded functions when clearing user functions

* symtab.h (symbol_table::fcn_info::fcn_info_rep::clear_user_function):
Call clear_autoload_function.

diff --git a/libinterp/corefcn/symtab.h b/libinterp/corefcn/symtab.h
--- a/libinterp/corefcn/symtab.h
+++ b/libinterp/corefcn/symtab.h
@@ -876,16 +876,18 @@ public:
         if (force || ! autoload_function.islocked ())
           autoload_function = octave_value ();
       }
 
       // We also clear command line functions here, as these are both
       // "user defined"
       void clear_user_function (bool force = false)
       {
+        clear_autoload_function (force);
+
         if (force || ! function_on_path.islocked ())
           function_on_path = octave_value ();
 
         if (force || ! cmdline_function.islocked ())
           cmdline_function = octave_value ();
       }
 
       void clear_mex_function (void)
